src/battle.ts

```diff
@@ -867,15 +866,16 @@ export class Side {
 		}
 		this.battle.scene.animSummon(pokemon, slot, true);
 	}
-	switchOut(pokemon: Pokemon, slot = pokemon.slot) {
+	switchOut(pokemon: Pokemon, kwArgs: KWArgs, slot = pokemon.slot) {
+		const effect = Dex.getEffect(kwArgs.from);
 		if (pokemon.lastMove !== 'batonpass' && pokemon.lastMove !== 'zbatonpass') {
 			pokemon.clearVolatile();
 		} else {
 			pokemon.removeVolatile('transform' as ID);
 			pokemon.removeVolatile('formechange' as ID);
 		}
-		if (pokemon.lastMove === 'uturn' || pokemon.lastMove === 'voltswitch') {
-			this.battle.log(['switchout', pokemon.ident], {from: pokemon.lastMove});
+		if (['uturn', 'voltswitch', 'flipturn'].includes(effect.id)) {
```

Comment:
Is there a way we can generalize this better? This is missing Parting Shot too, for example.

Reply:
how about this?
```js
if (pokemon.lastMove !== 'batonpass' && pokemon.lastMove !== 'zbatonpass') {
	this.battle.log(['switchout', pokemon.ident], {from: kwArgs.from});
}
```
`BattleTextParser` should automatically deal with the `from` effect not having its own switch out text (or 'from` not existing

Reply:
Your solution here looks good.